src/system_unix.cpp

```diff
@@ -170,28 +170,31 @@ SubProcess SubProcess::create(const std::string& run_as_user, const std::vector<
         close(reading_end_fd);
 
         SubProcess handle{writing_end_fd, pid};
-
-        if (not capabilities.empty()) {
-            // we need to keep caps, otherwise, we'll loose all our capabilities (except inherited)
-            if (system::keep_caps() == false) {
-                throw std::runtime_error("Keeping capabilities (SECBIT_KEEP_CAPS) failed");
+        try {
+            if (not capabilities.empty()) {
+                // we need to keep caps, otherwise, we'll loose all our capabilities (except inherited)
+                if (system::keep_caps() == false) {
+                    throw std::runtime_error("Keeping capabilities (SECBIT_KEEP_CAPS) failed");
+                }
             }
-        }
-
-        // Set real user for child process
-        auto error = system::set_real_user(run_as_user);
-        if (not error.empty()) {
-            throw std::runtime_error(fmt::format("Failed to set real user to: {}", run_as_user));
-        }
 
-        // Set capabilities for child process
-        if (not capabilities.empty()) {
-            error = system::set_caps(capabilities);
+            // Set real user for child process
+            auto error = system::set_real_user(run_as_user);
             if (not error.empty()) {
-                throw std::runtime_error(fmt::format("Failed to set capabilities: {}", error));
+                throw std::runtime_error(fmt::format("Failed to set real user to: {}", run_as_user));
             }
-        }
 
+            // Set capabilities for child process
+            if (not capabilities.empty()) {
+                error = system::set_caps(capabilities);
```

Comment:
we could just move this block into a separate function (would make things easier), and also, instead of throwing, we could again just return an error string

Reply:
done, removed the try/catch